### DISCUSSION TITLE
Light flash effect and restoring state

### DISCUSSION THREAD

**Discussion by SVH-Powel** [Koenkk/zigbee2mqtt#15457] [General]
Hi. I have recently gone from Deconz to Zigbee2Mqtt and are a happy user. Deconz has served me well for many years, but developed more and more issues the last year and they was apparently not able to fix it. So when all my Philips Hue Dimmers stopped working this summer, I decided to switch to Z2M. I have 140 devices on the zigbee network and everything has gone extremely well. The only problem I had was with the IKEA rollers which I took down from the wall and placed next to the coordinator for the pairing to work.

But here is my question: I am using Home Assistant and the service call light turn on with flash = short when to alert me. This has worked well with Deconz, but with Z2M, the light stay in on state after the flash. I vaguely remember that Deconz made some sort of functionality to restore state when turn on with flash was used.

You will probably tell me to check state and restore myself, but that is not easy. I am flashing a light group and I dont know the state of individual lights in this group. I dont even know which lights that are in the group (well, I do. But Home Assistant don't know)

So how can I flash the lights in a group with Z2M and restore state afterwards? Is it even possible?

**Comment by Hypfer** [comment]
I've just encountered the exact same situation and it seems that the issue is not z2m but Home Assistant.

Specifically, the effect parameter being part of `light.turn_on`, because that sends `{"state":"ON","effect":"breathe"}` if you only select the effect which z2m happily complies to do. If you manually publish just `{"effect":"breathe"}`, the light will do the breathe effect and then return to its previous state.

I suspect that the deconz integration might have a filter for that? Or maybe deconz itself does?
Either way, you can get the desired effect with no change to the state by using `mqtt.publish`.
Though this is probably something to raise with HA as bypassing the abstraction surely can't be the right thing to do.
